File: TestNodeJs/models/person.model.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class Person extends Model {
    static associate(models) {
      this.belongsTo(models.Family, {
        foreignKey: "family_id",
        as: "family",
      });

      this.belongsToMany(models.Asset, {
        through: "Person_Assets",
        foreignKey: "person_id",
        as: "assets",
        timestamps: false,
      });
    }
  }
  Person.init(
    {
      id: {
        type: DataTypes.INTEGER.UNSIGNED,
        allowNull: false,
        primaryKey: true,
        autoIncrement: true,
      },
      nama: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      jenis_kelamin: {
        type: DataTypes.STRING,
        allowNull: false,
      },
    },
    {
      sequelize,
      paranoid: true,
      modelName: "Person",
      tableName: "People",
      createdAt: "created_at",
      updatedAt: "updated_at",
      deletedAt: "deleted_at",
    }
  );
  return Person;
};

File: TestGolang/models/person.model.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Person struct {
	Id            uint           `gorm:"primaryKey,autoIncrement" json:"id"`
	Nama          string         `gorm:"not null" json:"nama"`
	Jenis_kelamin string         `gorm:"not null" json:"jenis_kelamin"`
	Family_id     uint           `gorm:"references:id"`
	Family        *Family        `json:"keluarga"`
	Asset         []*Asset       `gorm:"many2many:person_assets"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `gorm:"null" json:"deleted_at"`
}

func (Person) TableName() string {
	return "people"
}
